sponsorship_compassion/model/contracts.py

```diff
@@ -511,68 +481,57 @@ def fields_view_get(self, cr, user, view_id=None, view_type='form',
+        super(sponsorship_contract, self).on_change_partner_id()
         # Check if group_id is valid
-        if 'group_id' in res['value']:
-            if not self._is_a_valid_group(
-                    cr, uid, res['value']['group_id'], context):
-                del res['value']['group_id']
-
-        if ids:
-            contract = self.browse(cr, uid, ids[0], context)
-            if 'S' in type and contract.state == 'draft':
-                # If state draft correspondant_id=partner_id
-                res['value']['correspondant_id'] = partner_id
-        elif 'S' in type:
-            res['value']['correspondant_id'] = partner_id
+        if self.group_id:
+            if not self._is_a_valid_group(self.group_id.id):
+                self.group_id = False
 
-        return res
+        if 'S' in self.type and self.state == 'draft':
+            # If state draft correspondant_id=partner_id
+            self.correspondant_id = self.partner_id
+        elif 'S' in self.type:
+            self.correspondant_id = self.partner_id
 
     ##########################################################################
     #                            WORKFLOW METHODS                            #
     ##########################################################################
-    def contract_cancelled(self, cr, uid, ids, context=None):
-        res = super(sponsorship_contract, self).contract_cancelled(
-            cr, uid, ids, context)
+    @api.multi
+    def contract_cancelled(self):
+        res = super(sponsorship_contract, self).contract_cancelled()
 
-        sponsorship_ids = self.search(cr, uid, [
-            ('id', 'in', ids),
-            ('type', 'like', 'S')], context=context)
-        self._on_sponsorship_finished(cr, uid, sponsorship_ids, context)
+        sponsorships = self.filtered(lambda c: c.type == 'S')
+        self._on_sponsorship_finished(sponsorships)
 
         return res
 
-    def contract_terminated(self, cr, uid, ids, context=None):
-        res = super(sponsorship_contract, self).contract_terminated(
-            cr, uid, ids, context)
+    @api.multi
+    def contract_terminated(self):
+        res = super(sponsorship_contract, self).contract_terminated()
 
-        sponsorship_ids = self.search(cr, uid, [
-            ('id', 'in', ids),
-            ('type', 'like', 'S')], context=context)
-        self._on_sponsorship_finished(cr, uid, sponsorship_ids, context)
+        self._on_sponsorship_finished(self.filtered(lambda c: c.type == 'S'))
 
         return res
 
-    def contract_waiting_mandate(self, cr, uid, ids, context=None):
-        for contract in self.browse(cr, uid, ids, context):
+    @api.multi
+    def contract_waiting_mandate(self):
+        for contract in self:
             # Check that a child is selected for Sponsorship product
             if 'S' in contract.type and not contract.child_id:
-                raise orm.except_orm(
+                raise exceptions.Warning(

```

Comment:
I think we can completely remove this check as child_id is a required field when type is 'S'...